docs/advanced/tls_hardware_acceleration.md

```diff
@@ -0,0 +1,131 @@
+# mbed TLS Hardware Acceleration
+
+This document explains how to use hardware acceleration for various cryptographic primitives with [mbed TLS](https://github.com/ARMmbed/mbedtls) running on an embedded device.
+
+
+## Introduction
+
+### Why should I add hardware acceleration?
+
+Whether the application developer uses mbed TLS as a cryptographic library or uses it as a TLS stack, they will end up with time-consuming cryptographic routines in their application that will heavily impact response time. The platform that reduces this adverse effect will have an edge over platforms that do not (or do so to a lesser extent).
+
+You may want to add hardware acceleration in the following cases:
+
+
+- You can accelerate parts significantly with optimised assembly code.
+
+
+- Your processor has special instructions capable of accelerating cryptographic operations.
```

Comment:
Can we combine lines 15 and 18 onto the same line? They're almost the same thing.